### PR TITLE
Allow updating fields to null

### PR DESCRIPTION
**Pull request by eeSeeGee**
Signed-off-by: Aaron Young <aaron.young@gmail.com>

### CODE REVIEW

**Comment by sougou** [comment]
It just occurred to me that we need to do more work on this. This is because the lookup query will end up generating `where col = null`, which will always be false. Also, inserting `null` in a pk will fail.

What we can do is convert nulls to `0` (in lookup_internal.go). In the case of a string key, it will get treated as `"0"`, which should just work. However, this means that we can't allow a `null` and `0` for a unique lookup.

In other words, there's no clean way to support null as first class value, but the above solution seems to be a decent compromise.

**Comment by sougou** [comment]
The `null` issue was caused by the previous merge and not this one. So, I'll merge this, and we can fix it in a new PR.

**Comment by sougou** [comment]
This was what I was going to originally propose until I realized that inserts with null values will anyway fail because of primary key constraints.

**Comment by eeSeeGee** [comment]
The thing about null primary keys is they don't really matter for our use case because we use Hibernate. I'm also kind of of the mind that if you allow nulls in your primary key you deserve all the trouble you're asking for :)
